index.js

```diff
@@ -1,39 +1,171 @@
-const readline = require('readline');
-const readlineInterface = readline.createInterface(process.stdin, process.stdout);
+const readline = require("readline");
+const readlineInterface = readline.createInterface(
+	process.stdin,
+	process.stdout
+);
 
 function ask(questionText) {
-  return new Promise((resolve, reject) => {
-    readlineInterface.question(questionText, resolve);
-  });
+	return new Promise((resolve, reject) => {
+		readlineInterface.question(questionText, resolve);
+	});
 }
 
 // remember the StateMachine lecture
 // https://bootcamp.burlingtoncodeacademy.com/lessons/cs/state-machines
-let states = {
-  'roomOne': { canChangeTo: [ 'roomTwo' ] },
-  'roomTwo': { canChangeTo: [ 'roomThree' ] },
-  'roomThree': { canChangeTo: [ 'roomOne' ] }
+let states = { // I'm sure I need functions inside this for change the inventory of the room?
+	"182main": {
+		canChangeTo: ["foyer", "mr.mikes", "muddys"],
+		name: "182 Main St.",
+		welcomeMessage: `
+182 Main St. 
+You are standing on Main Street between Church and South Winooski.
+There is a door here. A keypad sits on the handle.
+On the door is a handwritten sign.
+
+`,
+		readSign: `
+
+The sign says "Welcome to Burlington Code Academy! Come on 
+up to the third floor. If the door is locked, use the code
+12345."
+
+`
+	},
+	"foyer": {
+		canChangeTo: ["182main", "stairs"],
```

Comment:
You're room objects are very similar to each other. It would probably be easier in the long run to set up a room class with a constructor to build these rooms, e.g.
```js
class Room {
  constructor(name, description, inventory) {
    this.name = name;
    this.description = description;
    this.inventory = inventory;
    this.pickup = (someVariable)=> {
    //do something with the variable 
    }
  }
}
```
It's more work to set up, but will make your program much easier to extend.

Reply:
I see exactly what the goal and direction is for this class but my brain can quite connect how it will generate my rooms and where I those rooms go and where I enter the information for them. For my understanding, what would you call this.pickup since it is a function inside the class? Why doesn't constructor (name, descript.,inventory) have pickup after it? Sorry lots of questions because this was the stuff that I hadn't seen enough of so I felt like I couldn't use it to finish my game.

Reply:
No worries, it's always best to build it in a way you know you can make it work before trying to get too fancy with it.  The pickup method doesn't need to be passed into the constructor since it's always going to be the same function no matter the room. You could then call it after creating a room by doing something like:
```js
const thisRoom = new Room('room', 'a generic room', [item1, item2])
thisRoom.pickup(item1) //to take item1
```
You can add as many of these methods onto the function as you want, and they will all work in the same way, and you will only have to write them the one time in the constructor.